docker/Dockerfile

```diff
@@ -37,11 +51,16 @@ RUN scripts/install_tools.sh
 RUN go generate && go build -tags production -o kjudge cmd/kjudge/main.go
 
 # Stage 3: Create awesome output image
-FROM ubuntu:jammy
+FROM base-ubuntu
 
-RUN apt-get update && DEBIAN_FRONTEND="noninteractive" TZ="Asia/Ho_Chi_Minh" apt-get install -y \
-    build-essential openjdk-11-jdk-headless fp-compiler python3.6 cgroup-lite python2.7 rustc golang libcap-dev # Compilers  \
-    openssl # For HTTPS support
+# Python PPA
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && apt-get install -y software-properties-common && add-apt-repository ppa:deadsnakes/ppa && \
```

Comment:
Per our talks can you cite some resources saying that Python 3.6 is the most adequate for competitive programming?
I personally don't enjoy holding packages of outdated Python from a 3rd-party PPA.

Reply:
Well it's Python so I guess it doesn't matter whether it's performance characteristics is fitting for CP. It's just most judges I see uses a version somewhere around 3.6. Updating to the newest Python might actually be a good choice. However, I think we should still notify contestants of its version.

Reply:
Ok, we can merge this for 0.8.

Reply:
As for how Python3.6 fare in terms of performance compared to other versions, I haven't found the data I talked about before. Probably will run pyperf tests tomorrow to see how the different versions actually performs. One thing to note, although it seems like 3.11 performed very well on the tests, but has not been adopted by popular libraries despite having been released for quite a while. Maybe I'll look into it tmr too.

Reply:
Uh, I guess there's no reason to keep python3.6, since it looks like for relevant tests, different versions of CPython are on par. The results are available at https://speed.python.org/comparison/. There are only results for 3.8-3.11, but I doubt that 3.6 will differ by a large margin. Also, it seems like even the deadsnakes PPA ditched support for 3.6 and there are no other reputable source for it, so uh... I guess we may consider switching to another package manager if we wish to support older Python distributions, or just simply don't support them.

Reply:
I simply bumped it to python3.10, to keep the build reproducible for the time being.